wiki/Client/Beatmap_editor/Menu/en.md

```diff
@@ -189,56 +181,54 @@ Places of interest.
+| Time Signature | Choose the [signature](/wiki/Music_theory/Time_signature) of the current timing point. For exotic timing, use the [`Timing Setup`](/wiki/Client/Beatmap_editor/Timing) window. |
+| Metronome Clicks | Toggle ticking sounds of the [metronome](/wiki/Client/Beatmap_editor/Timing#metronome) on the `Timing` tab. |
 
 ### Markers commands
 
 | Name | Description |
 | :-- | :-- |
-| Add Timing Section (`Ctrl` + `P`) | Add red offset (New BPM) |
-| Add Inheriting Section (`Ctrl` + `Shift` + `P`) | Add green offset (New adjuster) |
-| Reset Current Section | Reset the BPM setting of the red offset at the start of the section. Saving the beatmap without setting a new BPM will delete the red offset. |
-| Delete Timing Section (`Ctrl` + `I`) | Delete the red/green offset (delete one before the double line) |
-| Resnap Current Section | Resnap the current section with new current timing. |
+| Add Timing Section (`Ctrl` + `P`) | Add a new [uninherited (red) timing section](/wiki/Client/Beatmap_editor/Timing#uninherited-timing-point). |
+| Add Inheriting Section (`Ctrl` + `Shift` + `P`) | Add a new [inherited (green) timing section](/wiki/Client/Beatmap_editor/Timing#inherited-timing-point). |
+| Reset Current Section | Reset [BPM](/wiki/Music_theory/Tempo) and [offset](/wiki/Offset) of the current uninherited timing section, allowing to [re-time](/wiki/Beatmapping/Timing) it. Saving the beatmap will delete the sections without timing settings. |
+| Delete Timing Section (`Ctrl` + `I`) | Delete the current timing section (both inherited and uninherited). |
+| Resnap Current Section | [Resnap](/wiki/Beatmapping/Snapping) all hit objects in the current section according to the active beat snap divisor. |
 
 ### Timing Setup
 
 | Name | Description |
 | :-- | :-- |
-| Timing Setup... (F6) | [Timing Setup.](/wiki/Client/Beatmap_editor/Timing) |
+| Timing Setup... (`F6`) | Open the [`Timing and Control Points`](/wiki/Client/Beatmap_editor/Timing) window. |
 
 ### Special commands
 
 | Name | Description |
 | :-- | :-- |
-| Resnap All Notes | Resnap all the notes to their respective sections |
-| Move all notes in time.. | Move all the notes by #ms |
-| Recalculate Slider Lengths | Let osu! recalculate the length of sliders. (Best to use if bpm/slider velocity/slider velocity multiplier had changed.) **Caution:** Doing so can shorten none to all slider lengths! Check your map after use! |
-| Delete All Timing Sections | Purge timing. |
+| Resnap All Notes | Resnap all hit objects of the difficulty according to the active beat snap divisor. |
+| Move all notes in time.. | Move all hit objects by a given amount of milliseconds. |
+| Recalculate Slider Lengths | Resnap the [ends](/wiki/Gameplay/Hit_object/Slider/Slidertail) of all sliders to the nearest timeline tick, **potentially shortening them** (requires manual rechecking). Useful if the BPM or [slider velocity](/wiki/Gameplay/Hit_object/Slider/Slider_velocity) have been changed. |
```

Comment:
iirc it was agreed at some point to use "slider end" anyway (rc uses it)

when using "or" and both nouns are singular, it's still grammatically singular

```suggestion
| Recalculate Slider Lengths | Resnap [slider ends](/wiki/Gameplay/Hit_object/Slider/Slidertail) to the nearest timeline tick, **potentially shortening them** and requiring manual rechecking. Useful if the BPM or [slider velocity](/wiki/Gameplay/Hit_object/Slider/Slider_velocity) has been changed. |
```

Reply:
hmm okay